tests/e2e/monitoring/test_monitoring_on_negative_scenarios.py

```diff
@@ -83,14 +83,37 @@ def init_sanity(self):
     @pytest.fixture(autouse=True)
     def teardown(self, request, nodes):
         """
-        Restart nodes that are in status NotReady, for situations in
-        which the test failed before restarting the node after detach volume,
-        which leaves nodes in NotReady
+        Restart nodes that are in status NotReady or unschedulabe,
+        for situations in which the test failed in between restarting
+        or scheduling those nodes
 
         """
 
         def finalizer():
 
+            # Validate all nodes are schedulable
```

Comment:
@Akarsha-rai I believe this should be part of making nodes 'ready'. Why do you want to add this as part of a finalizer?

Reply:
On some of the tests(i.e, test_monitoring_after_draining_node_where_prometheus_hosted, etc), will mark node as unschedulable. If the test failed in between, to mark those node schedulabe again added this function on finalizer so that other tests won't have impact.